contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/schema/HiveDatabaseSchema.java

```diff
@@ -63,89 +58,38 @@ public Table getTable(String tableName) {
     return hiveSchema.getDrillTable(this.name, tableName);
   }
 
+  @Override
+  public Collection<Map.Entry<String, TableType>> getTableNamesAndTypes() {
+    ensureInitTables();
+    return tables.entrySet();
+  }
+
   @Override
   public Set<String> getTableNames() {
+    ensureInitTables();
+    return tables.keySet();
+  }
+
+  private void ensureInitTables() {
     if (tables == null) {
       try {
-        tables = Sets.newHashSet(mClient.getTableNames(this.name, schemaConfig.getIgnoreAuthErrors()));
-      } catch (final TException e) {
-        logger.warn("Failure while attempting to access HiveDatabase '{}'.", this.name, e.getCause());
-        tables = Sets.newHashSet(); // empty set.
+        tables = mClient.getTableNamesAndTypes(this.name, schemaConfig.getIgnoreAuthErrors());
+      } catch (TException e) {
+        logger.warn(String.format(
```

Comment:
Why `String.format`?

Reply:
It's invocation of ```warn(String msg, Throwable t)``` which means that stack trace won't be missed in logs. Using string with placeholders ```{}``` and ```warn(String format, Object... arguments)``` most probably will just call ```toString()``` on exception object and stack trace details won't be shown.

Reply:
I am ok about `e` instead of `e.getCause()`, but in both cases it is the last argument for `warn` method. So not sure what is the difference will a String formatting approach. However WARN level is used by default, users will not have benefit from placeholders. And this is not important. So you can keep your approach here.